core/src/Kokkos_CopyViews.hpp

```diff
@@ -2858,6 +2858,18 @@ inline void deep_copy(
   dst_value_type* dst_end   = dst.data() + dst.span();
   src_value_type* src_start = src.data();
   src_value_type* src_end   = src.data() + src.span();
+  if (((std::ptrdiff_t)dst_start == (std::ptrdiff_t)src_start) &&
+      ((std::ptrdiff_t)dst_end == (std::ptrdiff_t)src_end) &&
+      (dst.span_is_contiguous() && src.span_is_contiguous())) {
+    Kokkos::fence();
+#if defined(KOKKOS_ENABLE_PROFILING)
+    if (Kokkos::Profiling::profileLibraryLoaded()) {
+      Kokkos::Profiling::endDeepCopy();
+    }
+#endif
+    return;
+  }
```

Comment:
This is identical to https://github.com/dalg24/kokkos/blob/7623bdacbe639a389408e433283da6e0e2a0a79a/core/src/Kokkos_CopyViews.hpp#L1728-L1743 and looks pretty uncontroversial.